paper-api/src/main/java/org/bukkit/block/CreakingHeart.java

```diff
@@ -1,7 +1,60 @@
 package org.bukkit.block;
 
+import org.bukkit.Location;
+import org.bukkit.entity.Creaking;
+import org.jspecify.annotations.NullMarked;
+import org.jspecify.annotations.Nullable;
+
 /**
  * Represents a captured state of a creaking heart.
  */
+@NullMarked
 public interface CreakingHeart extends TileState {
+
+    /**
+     * Gets the creaking protecting this creaking heart.
+     *
+     * @return the creaking or null if this creaking heart don't have protector.
+     */
+    @Nullable
+    Creaking getCreaking();
+
+    /**
+     * Sets the creaking protecting this creaking heart.
+     *
+     * @param creaking the creaking or null for make this creaking heart don't have protector
+     * @throws IllegalArgumentException if the creaking passed it's in another world
+     */
+    void setCreaking(@Nullable Creaking creaking);
+
+    /**
+     * Gets the max distance between the Creaking Heart and the Creaking before to remove.
+     *
+     * @return the max distance
+     */
+    int getMaxDistanceForCreaking();
```

Comment:
I do not like this method name.
Especially when we also might add API for the creaking spawn distance.

`getProtectorSurvivalRange`
`setProtectorSurvivalRange`
`getProtectorSpawnRange`
`setProtectorSpawnRange`
*might* work, more peoples input tho pls.

Reply:
the lasts ones not like just becauuse that value its not for spawn.. its for limit the max distance where a creaking can "live"

maybe the first can work.. but not sure..

Reply:
The last ones were for the new API regarding the replacement for scaling, sorry xD
But yea, more people time chime in.

Reply:
getCreakingRemovalDistance?

Reply:
> getCreakingRemovalDistance?

not sounds bad... maybe can use that then (based in not another comments xd)